axon-framework/queries/query-dispatchers.md

```diff
@@ -227,5 +227,132 @@ commandGateway.sendAndWait(new RedeemCmd("gc1", amount));
 
    Since we subscribed to updates with the `println()` method, the update will be printed out once it is received.
 
+
+### Streaming queries
+
+The Streaming query allows a client to stream large database result sets. The Streaming query relies on reactive stream model, specifically the `Flux` type. Using streaming query, does not imply that reactive db drivers need to be used, users can hook to the stream and control the flow of data.
+
+
+Streaming query is flexible enough to be used with **any** query return type. That means that any return type that is not Flux will automatically be converted to Flux, based on that Flux will return single or multiple items.
+
+Query gateway provides nifty `streamingQuery` method to utilize streaming query easily. 
+It's simple to use, and requires just two parameters: query and expected response type class.
+`streamingQuery` method **is lazy**, meaning query is sent once Flux is subscribed to.
+
+
+It is also possible to utilize streaming query with `query` method: `queryGateway.query(query,  ResponseTypes.fluxOf(CardSummary.class))`.
+However, this approach is **eager** and less intuitive as it requires more boilerplate code.
+In this case `query` method will send query immediately and return CompletableFuture which returns a payload of type Flux, that needs to be subscribed to.
+
+```java
+@QueryHandler
+public List<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return cardRepository.findAll(); //1
+}
+        ...
+
+public Flux<CardSummary> consumer() {
+        return queryGateway.streamingQuery(query, CardSummary.class); //2
+}
+```
+
+1. We are querying the `cardRepository` for all the cards, that can potently return a large result set containing thousands of items.
+2. We are using the `queryGateway` to issue the query. If we used `multipleInstanceOf(CardSummary.class)` we would get large list transferred as single result message over the network, potentially causing a buffer overflow and max message size violation.
+Instead, we are using `streamingQuery(query, CardSummary.class)`, that will convert our response to Flux stream and chunk the result set into smaller messages of CardSummary.
+
+Natively, if we want fine-grained control of the producing stream we can use Flux as return type:
+
+```java
+@QueryHandler
+public Flux<CardSummary> handle(FetchCardSummariesQuery query) {
+        ...
+    return reactiveCardRepository.findAll(); 
+}
+```
+
+When using Flux as return type, we can control backpressure, stream cancellation and implement more complex features like pagination.
+
+> **Note**
```

Comment:
It is better to have "named notes," like with the one directly coming after this one. So, something like "Axon Server Version considerations".
Furthermore, it looks a bit messy to have two notes directly after one another. 
Hence I'd recommend to move this note to a different portion of this subsection.